eclipse/plugins/org.switchyard.tools.ui.editor/src/org/switchyard/tools/ui/editor/components/http/HttpProxyComposite.java

```diff
@@ -184,7 +186,12 @@ protected void doSetValue(Object value) {
                     final ProxyType proxy = (ProxyType) value;
                     hostValue.setValue(proxy.getHost());
                     passwordValue.setValue(proxy.getPassword());
-                    portValue.setValue(proxy.getPort());
+                    if (proxy.getPort() instanceof BigInteger) {
```

Comment:
we have 2 cases here:
- BigInteger (with the instanceof)
- String (with the cast)

maybe it will be simpler to always call just proxy.getPort().toString()
for each case:
- BigInteger --> it is what we are doing now
- String --> calling toString on String will provide the same result

it will avoid to introduce a if statement, what do you think?